### ISSUE TITLE
Registry Check fails in external package using Tensorstore as a dependency - Windows Python Wheel

### ISSUE THREAD

**Issue by sameeul** [google/tensorstore#101]
This is not exactly a tensorstore issue but in a  [package](https://github.com/sameeul/omepyrgen/tree/main) that I created using tensorstore as a dependency. I am missing something very subtle which I think any of the tensorstore devs can help me to identify.

I have a kvstore driver to read the `tiled-tiff` file (I followed the file kvstore driver). And then I have a tensorstore driver `ometiff` to read OME TIFF files.

I have created a Python library and a test executable that exercises the same code path for the mentioned drivers. In linux, both the python wheel and the executable works as expected (reads the OME Tiff files). 

But in Windows, the test executable works but the python wheel gives the following error message when trying to import the package. 
```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR

F0000 00:00:1684938523.025752   17456 registry.cc:35] Check failed: m

*** Check failure stack trace: ***
```

I think this registration check is failing in the kvstore driver that I wrote: [tiled_tiff](https://github.com/sameeul/omepyrgen/blob/main/src/ts_driver/tiled_tiff/tiled_tiff_key_value_store.cc)

I would really appreciate if anyone can take a look and give me some feedback on this.

To summarize:
[Test executable](https://github.com/sameeul/omepyrgen/blob/main/tests/main.cpp) works in both Linux and Windows
Python wheel works in Linux but fails in Windows.

**Comment by jbms** [comment]
The failing check is here: https://github.com/google/tensorstore/blob/c853880acbf03a06141f3817e56da34e9a089770/tensorstore/internal/metrics/registry.cc#L35

This may be triggered by your metric registrations here: https://github.com/sameeul/omepyrgen/blob/0d75ade9b05388aef76d50dab16e0da040e58ca3/src/ts_driver/tiled_tiff/tiled_tiff_key_value_store.cc#L79

But I don't see anything wrong there.

Note that the `Metric` type is a `Poly` alias, which is like `std::function` but supports multiple signatures.  I don't see any code path that could lead to it being a null (uninitialized) Poly.

A few things you could try to debug this issue:
- Try building a Windows wheel from the upstream tensorstore sources, without your modifications, but using the same build process.  This will confirm that the crash on Windows is due to your changes, rather than how you are building the Windows wheel.
- Try using a windows debugger to see what is causing the null Poly instance.  The stack trace will tell you the specific metric registration that is causing the issue.
- Try removing those metrics and their uses to see if it fixes the problem.

**Comment by sameeul** [comment]
Removing those metrics resolved the issue. Thanks!
